src/event.cpp

```diff
@@ -542,6 +546,20 @@ Event::EventStatus Event::status() const
     return static_cast<Event::EventStatus>(d->flags.status);
 }
 
+Event::EventIncomingStatus Event::incomingStatus() const
+{
+    const QString type = extraProperty(EVENT_PROPERTY_INCOMING_STATUS_ID).toString();
+    if (type == INCOMING_STATUS_MISSED) {
+        return Event::Missed;
```

Comment:
The existing missed calls won't be having the extra property so could use the flag for marking the event missed? It could use both extra property or the flag, but maybe could just keep the missed call state storage as it was to avoid redundant and potentially conflicting data?

Reply:
I'm doing the synchronisation in the setters:
- `setIsMissed()` is changing the status to Missed when true (left it unchanged otherwise),
- `setIncomingStatus()` is calling setIsMissed(status == Missed).

Like that, if I'm not mistaken, they should always be consistent. For older database entries, `setIsMissed()` is called from `DatabaseIOPrivate::readEventResult()`, which should set the proper incoming status.

Reply:
Ah, but I see your point of not saving "missed" as an extra property, not to be redundant and potentially faulty.

I've amended the commit accordingly, to save only the two new states.